src/lib.rs

```diff
@@ -456,9 +440,8 @@ fn lexicographical_topological_sort(
 #[pyfunction]
 #[text_signature = "(graph, /)"]
 fn graph_greedy_color(
-    py: Python,
     graph: &graph::PyGraph,
-) -> PyResult<PyObject> {
+) -> PyResult<HashMap<usize, usize>> {
```

Comment:
This is showing a 10-30% regression:

```
Benchmarks that have stayed the same:
       
       before           after         ratio
     [4d66ef43]       [bee8e8b4]
     <pyo3-hashbrown~3>       <pyo3-hashbrown>
      1.98±0.08ms       2.56±0.1ms    ~1.29  coloring.ColoringBenchmarks.time_graph_greedy_coloring(10000, 100)
       16.3±0.9?s      20.1±0.06?s    ~1.24  coloring.ColoringBenchmarks.time_graph_greedy_coloring(100, 10)
         379±20?s         454±30?s    ~1.20  coloring.ColoringBenchmarks.time_graph_greedy_coloring(1000, 10000)
       1.99±0.1ms      2.38±0.04ms    ~1.20  coloring.ColoringBenchmarks.time_graph_greedy_coloring(10000, 10)
         34.5±1?s         40.2±2?s    ~1.16  coloring.ColoringBenchmarks.time_graph_greedy_coloring(100, 1000)
          263±6ms          301±5ms    ~1.14  coloring.ColoringBenchmarks.time_graph_greedy_coloring(1000000, 10)
       19.8±0.4ms       22.6±0.8ms    ~1.14  coloring.ColoringBenchmarks.time_graph_greedy_coloring(100000, 100)
         273±10ms          311±3ms    ~1.14  coloring.ColoringBenchmarks.time_graph_greedy_coloring(1000000, 1000)
      1.48±0.03?s         1.68±0?s    ~1.14  coloring.ColoringBenchmarks.time_graph_greedy_coloring(1, 100)
       2.20±0.1ms      2.49±0.05ms    ~1.13  coloring.ColoringBenchmarks.time_graph_greedy_coloring(10000, 1000)
      2.24±0.03?s      2.53±0.01?s    ~1.13  coloring.ColoringBenchmarks.time_graph_greedy_coloring(10, 10)
         274±10ms          307±7ms    ~1.12  coloring.ColoringBenchmarks.time_graph_greedy_coloring(1000000, 100)
          283±3ms          313±2ms    ~1.11  coloring.ColoringBenchmarks.time_graph_greedy_coloring(1000000, 10000)
          308±9ms         341±10ms    ~1.11  coloring.ColoringBenchmarks.time_graph_greedy_coloring(1000000, 100000)
          266±4ms         293±10ms    ~1.10  coloring.ColoringBenchmarks.time_graph_greedy_coloring(1000000, 1)
      5.63±0.08ms      6.20±0.04ms    ~1.10  coloring.ColoringBenchmarks.time_graph_greedy_coloring(1000, 100000)
       22.0±0.4ms       24.0±0.6ms     1.09  coloring.ColoringBenchmarks.time_graph_greedy_coloring(100000, 10000)
       20.6±0.7ms         22.4±1ms     1.09  coloring.ColoringBenchmarks.time_graph_greedy_coloring(100000, 1000)
         16.2±1?s       17.6±0.5?s     1.09  coloring.ColoringBenchmarks.time_graph_greedy_coloring(100, 100)
      2.49±0.07ms      2.71±0.08ms     1.08  coloring.ColoringBenchmarks.time_graph_greedy_coloring(100, 100000)
       24.0±0.4ms       26.0±0.8ms     1.08  coloring.ColoringBenchmarks.time_graph_greedy_coloring(100000, 100000)
      2.09±0.06ms      2.21±0.03ms     1.06  coloring.ColoringBenchmarks.time_graph_greedy_coloring(10000, 10000)
       20.3±0.5ms       21.4±0.5ms     1.05  coloring.ColoringBenchmarks.time_graph_greedy_coloring(100000, 10)
          225±9?s         236±20?s     1.05  coloring.ColoringBenchmarks.time_graph_greedy_coloring(100, 10000)
         651±50ns         680±70ns     1.05  coloring.ColoringBenchmarks.time_graph_greedy_coloring(1, 1)
         696±30ms         720±10ms     1.03  coloring.ColoringBenchmarks.time_graph_greedy_coloring(1000000, 1000000)
         222±10?s         229±20?s     1.03  coloring.ColoringBenchmarks.time_graph_greedy_coloring(10, 10000)
         44.9±1ms       46.1±0.5ms     1.03  coloring.ColoringBenchmarks.time_graph_greedy_coloring(10, 1000000)
      24.7±0.09?s         25.1±1?s     1.01  coloring.ColoringBenchmarks.time_graph_greedy_coloring(10, 1000)
       8.02±0.1ms       8.07±0.3ms     1.01  coloring.ColoringBenchmarks.time_graph_greedy_coloring(10000, 100000)
      9.93±0.05?s      9.98±0.03?s     1.01  coloring.ColoringBenchmarks.time_graph_greedy_coloring(1, 1000)
      2.38±0.09ms       2.39±0.1ms     1.00  coloring.ColoringBenchmarks.time_graph_greedy_coloring(10, 100000)
          930±2?s          932±2?s     1.00  coloring.ColoringBenchmarks.time_graph_greedy_coloring(1, 100000)
      92.8±0.09?s       92.9±0.2?s     1.00  coloring.ColoringBenchmarks.time_graph_greedy_coloring(1, 10000)
         291±20ms          290±8ms     1.00  coloring.ColoringBenchmarks.time_graph_greedy_coloring(100000, 1000000)
       76.9±0.5ms       74.4±0.9ms     0.97  coloring.ColoringBenchmarks.time_graph_greedy_coloring(100, 1000000)
          734±7ns         710±90ns     0.97  coloring.ColoringBenchmarks.time_graph_greedy_coloring(1, 10)
       13.1±0.1ms      12.6±0.09ms     0.96  coloring.ColoringBenchmarks.time_graph_greedy_coloring(1, 1000000)
          256±3ms          246±2ms     0.96  coloring.ColoringBenchmarks.time_graph_greedy_coloring(10000, 1000000)
         269±20ms          248±9ms     0.92  coloring.ColoringBenchmarks.time_graph_greedy_coloring(1000, 1000000)

Benchmarks that have got worse:
       
       before           after         ratio
     [4d66ef43]       [bee8e8b4]
     <pyo3-hashbrown~3>       <pyo3-hashbrown>
+         147±4?s        192±0.5?s     1.31  coloring.ColoringBenchmarks.time_graph_greedy_coloring(1000, 10)
+      1.91±0.1?s      2.40±0.02?s     1.25  coloring.ColoringBenchmarks.time_graph_greedy_coloring(10, 1)
+      13.2±0.6?s      15.8±0.03?s     1.20  coloring.ColoringBenchmarks.time_graph_greedy_coloring(100, 1)
+         157±1?s          183±6?s     1.17  coloring.ColoringBenchmarks.time_graph_greedy_coloring(1000, 1000)
+      4.01±0.2?s      4.65±0.01?s     1.16  coloring.ColoringBenchmarks.time_graph_greedy_coloring(10, 100)
+      19.2±0.8ms       22.1±0.3ms     1.15  coloring.ColoringBenchmarks.time_graph_greedy_coloring(100000, 1)
+     1.92±0.06ms      2.20±0.03ms     1.15  coloring.ColoringBenchmarks.time_graph_greedy_coloring(10000, 1)
+         128±8?s          146±9?s     1.15  coloring.ColoringBenchmarks.time_graph_greedy_coloring(1000, 1)
+         227±1?s          251±2?s     1.10  coloring.ColoringBenchmarks.time_graph_greedy_coloring(1000, 100)

SOME BENCHMARKS HAVE CHANGED SIGNIFICANTLY.
PERFORMANCE DECREASED.
```

from https://github.com/mtreinish/retworkx-bench which is a bit surprising since this should be more or less the same, it just moves the conversion loop into the ffi function that `#[pyfunction]` generates for us. But if it's that much of a difference it's probably worth reverting this before merging